command/agent.go

```diff
@@ -332,14 +333,29 @@ func (c *AgentCommand) Run(args []string) int {
 	if config.Cache != nil && len(config.Cache.Listeners) != 0 {
 		cacheLogger := c.logger.Named("cache")
 
+		originalEnvVaultAgentAddress := os.Getenv(api.EnvVaultAgentAddress)
+		os.Setenv(api.EnvVaultAgentAddress, "")
```

Comment:
This dance of temporarily unsetting `VAULT_AGENT_ADDR` seems weird to me. I wonder if the `api.Client` needs to be aware of both `VAULT_AGENT_ADDR` and `VAULT_ADDR` env vars if only only one is considered. I think we can use a single env var, and it simply means the address that the client will talk to, whether it's the server or a proxy.

Reply:
We've been through this idea. The problem with that is when both agent and client are on the same machine, agent wouldn't know how to reach vault as the single env var will be used to both client pointing to agent and agent pointing to Vault.

Reply:
Actually I can see why we might need the two, marking as resolved.

Reply:
This code seems super weird and I don't understand the explanation.

Reply:
@jefferai Basically, there are 2 env vars `VAULT_ADDR` and `VAULT_AGENT_ADDR`. `VAULT_AGENT_ADDR` is used by the clients to point to the agent. `VAULT_ADDR` is used by agent to point to Vault server. If the client created by the agent refers to `VAULT_AGENT_ADDR` it is essentially a loop back. So, we suppress that so that agent always only considers `VAULT_ADDR` and points to Vault server.

Reply:
Why not just make those separate fields in the client so that they can both be parsed and the correct one can be used?

Reply:
Maybe instead you can call `DefaultConfig()` and override the Address?

Reply:
I think we need to support setting the server's address in the config file before GA anyway, so we will need to do this eventually.

Reply:
I took both of your input into account and used that as the way forward. I have it working. I'll test it a bit more and run the tests on this before pushing. Stay put. I am hoping to make this PR into 1.1 beta.

Reply:
Fixed in 8da561ae726cd385be2fa8eefc2495868d494bd2

Reply:
The reason for not adding support for taking in Vault's address from config file is to decide if that address is applicable for agent cache alone or if it is applicable for the auto-auth case well. Currently auto-auth is using env vars just like any other CLI command and the same reasoning was carried over to the cache feature as well.